docs/api-gateway-v2/03-04-migration.md

```diff
@@ -0,0 +1,144 @@
+
+```yaml
+authentication:
+- type: JWT
+  jwt:
+    issuer: https://dex.kyma.local
+    jwksUri: http://dex-service.kyma-system.svc.cluster.local:5556/keys
+    triggerRule:
+      excludedPaths:
+      - exact: /exact/path/to/resource.jpg
+      - exact: /no/auth/needed/resource.html
+- type: JWT
+  jwt:
+    issuer: https://auth.kyma.local
+    jwksUri: http://auth-service.kyma-system.svc.cluster.local:5556/keys
+    triggerRule:
+      excludedPaths:
+      - prefix: /pref/
+      - exact: /no/auth/needed/resource.html
+```
+
+Here is the ApiRule configuration which enforces the same authentication policies:
+
+```yaml
+rules:
+  - path: /no/auth/needed/resource.html
+    methods: ["GET", "POST", "PUT", "DELETE"]
+    accessStrategies:
+    - handler: allow
+  - path: /pref/.*
+    methods: ["GET", "POST", "PUT", "DELETE"]
+    accessStrategies:
+    - handler: jwt
+      config:
+        trusted_issuers:
+        - "https://dex.kyma.local"
+        jwks_urls:
+        - "http://dex-service.kyma-system.svc.cluster.local:5556/keys"
+  - path: /exact/path/to/resource.jpg
+    methods: ["GET", "POST", "PUT", "DELETE"]
+    accessStrategies:
+    - handler: jwt
+      config:
+        trusted_issuers:
+        - "https://auth.kyma.local"
+        jwks_urls:
+        - "http://auth-service.kyma-system.svc.cluster.local:5556/keys"
+  - path: (?!/pref/.*)(?!/exact/path/to/resource.jpg)/.*
+    methods: ["GET", "POST", "PUT", "DELETE"]
+    accessStrategies:
+    - handler: jwt
+      config:
+        trusted_issuers:
+        - "https://dex.kyma.local"
+        - "https://auth.kyma.local"
+        jwks_urls:
+        - "http://dex-service.kyma-system.svc.cluster.local:5556/keys"
+        - "http://auth-service.kyma-system.svc.cluster.local:5556/keys"
+```
+
+The important part to notice is the last path configuration containing regex with a negative lookahead. It is used to exclude paths handled by other path settings from the `/.*` path, as there can't be two different configurations for a single path. There is one exception to that, which applies only if the same `excludedPaths` element is present on all `authentication` settings, so the specific path doesn't require any authentication at all. In that case, a rule with `handler: allow` should be created, and the path doesn't have to be excluded using a negative lookahead.
```

Comment:
```suggestion
Note that the last path configuration contains a regular expression with a negative lookahead. It is used to exclude paths handled by other path settings from the `/.*` path, as there can't be two different configurations for a single path. There is one exception to that, which applies only if the same `excludedPaths` element is present on all `authentication` settings, so the specific path doesn't require any authentication at all. In that case, a rule with `handler: allow` should be created, and the path doesn't have to be excluded using a negative lookahead.
```

> It is used to exclude paths handled by other path settings from the `/.*` path, as there can't be two different configurations for a single path.

This sentence might sound super confusing to regex greenhorns (like me). I think I get your point, but I'm not sure.

Reply:
I would replace the part: "as there can't be two different configurations for a single path" with: "as the path definitions in an APIRule **must** be unique."